src/c/types/array.c

```diff
@@ -43,24 +52,21 @@ bool ucdr_serialize_array_byte_2(ucdrBuffer* ub, const ucdrEndianness endianness
     uint32_t array_size = size * data_size;
     uint32_t alignment = ucdr_buffer_alignment(ub, sizeof(uint16_t));
 
-    if(ucdr_check_buffer(ub, alignment + array_size))
-    {
-        ub->iterator += alignment;
-        if(UCDR_MACHINE_ENDIANNESS == endianness)
-        {
-            memcpy(ub->iterator, array, array_size);
+    ub->iterator += alignment;
 
-            ub->iterator += array_size;
-            ub->last_data_size = data_size;
-        }
-        else
+    if(UCDR_MACHINE_ENDIANNESS == endianness)
+    {
+        ucdr_serialize_continuous_memory(ub, (uint8_t*)array, array_size);
+        ub->last_data_size = data_size;
+    }
+    else
+    {
+        for(uint32_t i = 0; i < size; i++)
         {
-            for(uint32_t i = 0; i < size; i++)
-            {
-                ucdr_serialize_byte_2(ub, endianness, array + i);
-            }
+            ucdr_serialize_byte_2(ub, endianness, array + i);
```

Comment:
This function will check the buffer inside of it, no previous checks are necessary